OpenRA.Mods.Common/Traits/Pluggable.cs

```diff
@@ -23,19 +25,29 @@ public class PluggableInfo : ITraitInfo, UsesInit<PlugsInit>
 		[Desc("Conditions to grant for each accepted plug type.")]
 		public readonly Dictionary<string, string> Conditions = null;
 
+		[Desc("Conditions to grant for each accepted plug type.")]
+		public readonly Dictionary<string, BooleanExpression> Requirements = new Dictionary<string, BooleanExpression>();
+
 		[GrantedConditionReference]
 		public IEnumerable<string> LinterConditions { get { return Conditions.Values; } }
 
+		[ConsumedConditionReference]
+		public IEnumerable<string> ConsumedConditions
+		{
+			get { return Requirements.Values.SelectMany(r => r.Variables.Except(Conditions.Values)); }
```

Comment:
It's valid for the same trait to both grant and consume the same condition.  If said trait was the only granter and consumer (not useful in this trait, but could be in others) then the linter would fail if it weren't listed in both.